source/common/upstream/utility.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+
+namespace Envoy {
+namespace Upstream {
+
+/**
+ * Utility functions for upstream.
+ */
+class Utility {
+public:
+  /**
+   * Resolve the cluster info.
+   * @param decoder_callbacks supplies the decoder callback of filter.
+   * @param cm supplies the cluster manager of the filter.
+   */
+  static Upstream::ClusterInfoConstSharedPtr
+  resolveClusterInfo(Http::StreamDecoderFilterCallbacks* decoder_callbacks,
```

Comment:
To me, upstream is a better place to hold it. It's not http filter specific, any filter which can convert a downstream protocol to sort of upstream may use it. Downstream protocol is not necessary to be http.